### PR TITLE
Add support for Model accessors to be called

### LINKED ISSUE #256: Mutating translated attributes.
Currently, Eloquent's Accessors won't be called on translated attributes. I checked up the translatable trait, and saw that it overrides `getAttribute()` method call. So, in order to make eloquent's getmutators work again, I think we should change the behavior of `getAttribute()` on translatable trait.
Here is the portion of change, 

```
if ($this->isTranslationAttribute($key)) {
        if ($this->getTranslation($locale) === null) {
           return;
        }

       if ($this->hasGetMutator($key)) {
            return $this->mutateAttribute($key, $this->getTranslation($locale)->$key);
       }
            
     return $this->getTranslation($locale)->$key;
}
```

The question is, is pull requests for something like this, allowed?

### PR DESCRIPTION
**Pull request by easteregg**
This fixes #256 .

### CODE REVIEW

File: src/Translatable/Translatable.php [hunk 1]
```diff
@@ -162,6 +162,10 @@ public function getAttribute($key)
                 return;
             }
 
+            if ($this->hasGetMutator($key)) {
+                return $this->mutateAttribute($key, $this->getTranslation($locale)->$key);
```
Comment (dimsav) [inline src/Translatable/Translatable.php]: Why do you call call `$this->mutateAttribute()` instead of `$this->getAttributeValue($key)`?

**Comment by dimsav** [comment]
Thanks a lot for this PR! Can you write a test that these changes will fix?

File: src/Translatable/Translatable.php (hunk 1 above)
Comment (easteregg) [inline src/Translatable/Translatable.php]: Thank you for the reply, @dimsav .

`$this->getAttributeValue($key)` will check the Model's `$attribute` array. and does not get the value from the translation. then internally checks if the `$key` has a mutator, then calls the mutator with the Model's `$value` (which is not translated.).

So calling `$this->getAttributeValue($key)` does not mutate translated attributes. 

But, you are right, using the `$this->getAttributeValue($key)` makes more sense, so I will change the code (by checking if `$key` has a mutator, then push it to `$attributes` before calling `$this->getAttributeValue($key)`.

**Comment by easteregg** [comment]
I will write the test ASAP for this.

**Comment by easteregg** [comment]
Ok, the tests are passing. Do you want me to squash commits? 

Oh, and BTW, I've seen the tests are extremely slow, take a look at this [branch](https://github.com/easteregg/laravel-translatable/tree/sqlite-unittesting/tests) which I tried to run the tests as in memory sqlite, and it is way faster than mysql.

**Comment by dimsav** [comment]
Thanks @easteregg! 

Indeed the tests are slow. From my experience running tests in sqlite is a bad practice because some mysql errors are not shown.

**Comment by dimsav** [comment]
The changes were just released. There are interesting techniques we could use to speedup mysql in tests. We can do this at a later moment.

**Comment by easteregg** [comment]
Thank you for your answer. I guess this issue is closed now.

**Pull request merged**